pkg/api/addons.go

```diff
@@ -243,16 +251,19 @@ func (cs *ContainerService) setAddonsConfig(isUpdate bool) {
 
 	defaultAzureNetworkPolicyAddonsConfig := KubernetesAddon{
 		Name:    AzureNetworkPolicyAddonName,
-		Enabled: azureNetworkPolicyAddonEnabled(o),
+		Enabled: to.BoolPtr(o.KubernetesConfig.NetworkPlugin == NetworkPluginAzure && o.KubernetesConfig.NetworkPolicy == NetworkPolicyAzure),
 		Containers: []KubernetesContainerSpec{
 			{
-				Name: AzureNetworkPolicyAddonName,
+				Name:  AzureNetworkPolicyAddonName,
+				Image: "mcr.microsoft.com/containernetworking/azure-npm:v1.0.18",
```

Comment:
Moved the container image default definition for this addon into this default assignment flow, out of the static definition in the yaml.